src/libraries/System.Private.CoreLib/src/System/IO/TextReader.cs

```diff
@@ -19,7 +19,7 @@ namespace System.IO
     // There are methods on the Stream class for reading bytes.
     public abstract partial class TextReader : MarshalByRefObject, IDisposable
     {
-        public static readonly TextReader Null = new NullTextReader();
+        public static readonly TextReader Null = StreamReader.Null;
```

Comment:
Overall the code looks good, but for some reason the tests are failing on **mono** (the CI has not finished so I am not 100% sure that it's mono bug):

```log
  Discovering: System.IO.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.IO.Tests (found 694 of 700 test cases)
  Starting:    System.IO.Tests (parallel test collections = on, max threads = 6)
    System.IO.Tests.NullTests.TestCanceledNullTextReaderAsync(input: null) [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.OperationCanceledException)
      Actual:   typeof(System.NullReferenceException): Object reference not set to an instance of an object
      ---- System.NullReferenceException : Object reference not set to an instance of an object
      Stack Trace:
        /_/src/libraries/System.IO/tests/Stream/Stream.NullTests.cs(184,0): at System.IO.Tests.NullTests.<>c__DisplayClass11_0.<<TestCanceledNullTextReaderAsync>b__0>d.MoveNext()
        --- End of stack trace from previous location ---
        ----- Inner Stack Trace -----
        /_/src/libraries/System.IO/tests/Stream/Stream.NullTests.cs(184,0): at System.IO.Tests.NullTests.<>c__DisplayClass11_0.<<TestCanceledNullTextReaderAsync>b__0>d.MoveNext()
        --- End of stack trace from previous location ---
      Assert.Throws() Failure
```

Is there any chance you could build it and try to repro locally? https://github.com/dotnet/runtime/blob/main/docs/workflow/building/mono/README.md

I have no idea why it fails only on Mono, but I guess it's caused by the fact that two out of three arugments provided by `NullReaders` refer to the same object instance.


https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.IO/tests/Stream/Stream.NullTests.cs#L197-L203

Reply:
I believe it's actually 3 out of 3 now. `StringReader.Null` always inherited from the base class, so it was the same object as `TextReader.Null`. But now they all are the same object. 

I could throw on a `.Distinct()` to ensure that all the properties are read, but that only unique objects are tested, but I'm not sure if that would stop `null` being passed as the input to the test method.

Reply:
Just had a thought that perhaps this is a static field initialisation problem (only on mono)? If the base class static fields have to be initialised before the derived class, then `TextReader.Null` would be set to `null`, which would cause the test failure. The fix would be to make `NullStreamReader` internal and simply construct a second instance in `TextReader.Null`. (`StreamReader.Null` technically wouldn't need to be a second instance, but could be `= (StreamReader) TextReader.Null` but that feels unnecessarily brittle to me.)

Reply:
> I could throw on a `.Distinct()`

That didn't fix the broken builds; I'll move ahead with the static field theory. (The `Distinct()` is still useful to avoid unnecessary testing.)